### PR TITLE
WIP: Add Mercurial

### PR DESCRIPTION
**Pull request by jakirkham**
Add a recipe to build Mercurial. Generated with `conda skeleton pypi` and updated respectively.

I did not prefix this with `python-`. 😮 I'm hoping that is not necessary as this is original library. However, we can discuss this.

### CODE REVIEW

**Comment by jakirkham** [comment]
On Linux it basically wants `unzip` for the tests. So, we either need to accept this PR ( https://github.com/conda-forge/docker-images/pull/20 ) or add it to `yum_requirements.txt`. This seems to be causing the bulk (all?) of the Linux failures.

**Comment by jakirkham** [comment]
The paths on Windows look funky and it seems to be unable find `hg`. ( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3051/job/8k5rv37ljmsew5r1 )

It looks like the tests won't run on Python 3.4. Should we still build there? ( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3051/job/r4e0x9q3ayi035l5 )

With Python 3.5, it seems to hit some sort of Python bug trying to run the tests. ( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.3051/job/jb4e36abg39wnut4 )

Basically, I'm thinking we should skip tests on Windows as this is a total mess.

**Comment by jakirkham** [comment]
On Mac it looks like we need a newer `git` than what Travis CI has (*I think*). ( https://travis-ci.org/conda-forge/staged-recipes/jobs/136022150 )

**Comment by jakirkham** [comment]
Toggling for AppVeyor.

**Comment by jakirkham** [comment]
Looks like AppVeyor has been having some trouble for ~20mins. Will let this sit for now.

**Comment by patricksnape** [comment]
Looks like the tests don't run on Windows: https://www.mercurial-scm.org/wiki/WritingTests

**Comment by jakirkham** [comment]
So, trying to figure out how to run the tests on Windows is a waste. Thanks for pointing that out.

When I ran the full test suite previously, I was getting all sorts of crazy failures.

Eventually resolved all but one on Linux. This test appears to be [recent]( https://www.mercurial-scm.org/pipermail/mercurial-devel/2016-January/078677.html ); so, may still have some kinks to work out. That or docker containers and Mercurial have weird interaction. Both are possible.

On Travis CI, we were having some issues where it was recommending some very lax settings to resolve the failures. Not sure how I feel about that solution. :/

In general, the tests were taking a very long time. Given the bulk of Mercurial is Python, Not really convinced that running the full test suite is the right strategy. Would much rather test the few portions that have compiled code. Though haven't decided how that should be done yet.

**Comment by jakirkham** [comment]
Replaced by PR ( https://github.com/conda-forge/staged-recipes/pull/1170 ).
